### PR TITLE
Add SetFilterParents post message

### PR DESCRIPTION
**Pull request by yanlipnican**
- New postmessage API command and event types

JIRA: BB-2436

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1285/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2875/ : FAILURE in 2m 15s

File: libs/sdk-embedding/src/iframe/kd.ts [hunk 1]
```diff
@@ -1023,4 +1039,131 @@ export namespace EmbeddedKpiDashboard {
...
+        /**
+         * Multiple filters with same id in single command
+         */
+        DuplicateFilters = "duplicateFilters",
+
+        /**
+         * Multiple parents with same id in single filter
+         */
+        DuplicateParents = "duplicateParents",
+    }
```
Comment (yanlipnican) [inline libs/sdk-embedding/src/iframe/kd.ts:1160]: Error codes for other messages are in AD/KD for some reason so they are not accessible to SDK users. This is much better approach IMO

File: libs/sdk-embedding/src/iframe/kd.ts [hunk 2]
```diff
@@ -1023,4 +1039,131 @@ export namespace EmbeddedKpiDashboard {
...
+        filters: ISetFilterParentsItem[];
+    }
+
+    /**
+     * @public
+     */
+    export type SetFilterParentsCommandData = IGdcKdMessageEnvelope<
+        GdcKdCommandType.SetFilterParents,
+        ISetFilterParentsDataBody
+    >;
```
Comment (lupko) [inline libs/sdk-embedding/src/iframe/kd.ts:1086]: I think you are missing the SetFilterParentCommand type def.

`SetFilterParentsCommand =  IGdcKdMessageEvent<..., ...>;`

Comment (lupko) [reply libs/sdk-embedding/src/iframe/kd.ts:1086]: also, when you add the command type, add contractual documentation for the command. what it returns on success, what on error etc

File: libs/sdk-embedding/src/iframe/kd.ts [hunk 3]
```diff
@@ -1023,4 +1039,131 @@ export namespace EmbeddedKpiDashboard {
...
+        filter: SetFilterParentsItemFilter;
+        parents: ISetFilterParentsItemParent[];
+    }
+
+    /**
+     * @public
+     */
+    export interface ISetFilterParentsDataBody {
+        filters: ISetFilterParentsItem[];
+    }
```
Comment (lupko) [inline libs/sdk-embedding/src/iframe/kd.ts:1078]: please look at these from consumer perspective and add TSDoc for the various types and properties. the documentation should give user enough information to know 'what to put where', how to construct the command body.

or look at it from perspective where we will be generating fancy public API documentation from the code. The API documentation should be verbose enough for consumers to understand how to construct the payload.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1304/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2900/ : FAILURE in 3m 22s

File: libs/sdk-embedding/src/iframe/kd.ts [hunk 4]
```diff
@@ -1023,4 +1039,236 @@ export namespace EmbeddedKpiDashboard {
...
+     * @public
+     */
+    export interface ISetFilterParentsAttributeFilter {
+        attributeFilter: {
+            displayForm: ObjRef;
+        };
+    }
+
+    /**
+     * Type that represents filter in dashboard. At the moment it can only be a attribute filter
```
Comment (lupko) [inline libs/sdk-embedding/src/iframe/kd.ts:1055]: `an attribute filter`

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1305/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2902/ : FAILURE in 3m 30s

File: libs/sdk-embedding/src/iframe/kd.ts [hunk 5]
```diff
@@ -1023,4 +1039,236 @@ export namespace EmbeddedKpiDashboard {
...
+     *
+     * @public
+     */
+    export type SetFilterParentsItemFilter = ISetFilterParentsAttributeFilter;
+
+    /**
+     * Type that represents filter connection to it's parent.
```
Comment (lupko) [inline libs/sdk-embedding/src/iframe/kd.ts:1062]: `its parent`

File: libs/sdk-embedding/src/iframe/kd.ts [hunk 6]
```diff
@@ -1023,4 +1039,236 @@ export namespace EmbeddedKpiDashboard {
...
+     *
+     * @public
+     */
+    export interface ISetFilterParentsItemParent {
+        /**
+         * Parent is filter that is present in dashboard.
```
Comment (lupko) [inline libs/sdk-embedding/src/iframe/kd.ts:1068]: `in dashboard` -> `on a dashboard`

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1306/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2903/ : FAILURE in 3m 30s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1307/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2904/ : SUCCESS in 13m 47s

**Review by lupko — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/466/ : FAILURE in 2m 02s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/725/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2915/ : SUCCESS in 20m 00s

**Pull request merged**